Resources/config/form_types.yml

```diff
@@ -0,0 +1,5 @@
+services:
+    pim_custom.form.type.custom_entity:
```

Comment:
pim_custom_entity.form.type.custom_entity or pim_custom_entity.form.type.reference_data ? As you prefer but we should rename classes accordingly